### DISCUSSION TITLE
Brownie dev is speeder than hardât

### DISCUSSION THREAD

**Discussion by wilonweb** [smartcontractkit/full-blockchain-solidity-course-py#1517] [Q&A] [answered]
Hello,

I studied the algorithmic in js and I learn react for the front.  in parallel I learn web3 with the python brownie framework (the course of Patrick Collins).  I'm on the 5th hour and I'm hesitant to drop it and switch to the one with hardHat or continue it. 
Is coding a smart contract with python is faster?  (because the video lassés 16 hours and the other 32) 
what are the differences between the 2 technologies?  Is there an advantage over one technology over another ?

you will advise me to finish the one in python before starting again with the one in JS or start directly the one in JS (because I know JS better than Python)

**Comment by cromewar** [comment, marked as answer]
Hello @NinjaTurtles-cloud 
The Js Version is longer because Patrick create more optional front-end projects to connect with the smart contracts, and some lesson like Daos are not present on the Python version.

So, regarding to which framework is "better", the answer is none of them. I personally prefer python so I work almost exclusively with brownie, but hardhat is great too. So, is your choice.

Patrick just made an article for help you to decide: https://medium.com/@patrick.collins_58673/learn-blockchain-development-in-one-of-these-two-videos-c7db210b98fd

I hope this info might help, cheers.

**Reply by wilonweb** [reply]
Thanks for the share
The article motivate me to stay  on brownie.
